tap_salesforce/client.py

```diff
@@ -231,6 +255,10 @@ def _make_request(self, method, path, data=None, params=None) -> requests.Respon
             method, url, headers=headers, params=params, data=data
         )
 
+        json = resp.json()
+        if isinstance(json, list) and isinstance(json[0], dict) and json[0]['errorCode'] == 'QUERY_TIMEOUT':
+            raise QueryTimeoutError()
```

Comment:
when ever we do very specific error handling like this, do provide the example payload of how the errors look as a comment somewhere.

Creating a custom error at this level is actually fine, but we should try to adhere to the current structure. There's already an `exceptions.py` file that contains the exceptions for different cases - I'd add a new one there and then only look at the error if the `resp.status_code` is `!= 2xx` - meaning not relying on the `resp.raise_for_status()`.

Something 

> 

> ala:

```python
if 200 <= resp.status_code < 300
```

Reply:
took your suggestion here: [2f297a8](https://github.com/dreamdata-io/tap-salesforce/pull/21/commits/2f297a84278436891eee7b039217aa70b307f7c4)